packages/typespec-ts/test/unit/clientFactoryGenerator.spec.ts

```diff
@@ -391,6 +391,61 @@ describe("Client Factory generation", () => {
       }
     });
 
+    it("should generate TokenCredential if scope is empty", async () => {
+      const factoryFile = await emitClientFactoryFromTypeSpec(
+        `
+        @useAuth(
+          OAuth2Auth<[{
+            type: OAuth2FlowType.implicit,
+            authorizationUrl: "https://login.microsoftonline.com/common/oauth2/v2.0/authorize",
+            scopes: []
+          }]>)
+        @service( {title: "PetStoreClient"})
+        namespace PetStore;
+      `,
+        false,
+        false
+      );
+
+      assert.ok(factoryFile);
+      // console.log(factoryFile!.content);
+      assertEqualContent(
+        factoryFile!.content,
+        `
+      import { getClient, ClientOptions } from "@azure-rest/core-client";
+      import { logger } from "./logger";
+      import { TokenCredential } from "@azure/core-auth";
+      import { testClient } from "./clientDefinitions";
+
+      /**
+       * Initialize a new instance of \`testClient\`
+       * @param endpoint - The parameter endpoint
+       * @param credentials - uniquely identify client credential
+       * @param options - the parameter for all optional parameters
+       */
+      export default function createClient(endpoint: string, credentials: TokenCredential, options: ClientOptions = {}): testClient {
+        const baseUrl = options.baseUrl ?? \`\${endpoint}\`;
+        
+        const userAgentInfo = \`azsdk-js-test-rest/1.0.0-beta.1\`;
+        const userAgentPrefix = options.userAgentOptions && options.userAgentOptions.userAgentPrefix ? \`\${options.userAgentOptions.userAgentPrefix} \${userAgentInfo}\`: \`\${userAgentInfo}\`;;
+        options = {
+            ...options,
+            userAgentOptions: {
+              userAgentPrefix
+            },
+            loggingOptions: {
+              logger: options.loggingOptions?.logger ?? logger.info
+            }
+        };
+
+        const client = getClient(baseUrl, options) as testClient;
+
+        return client;
```

Comment:
I mean the logic where you handle default credential scopes in the case of typespec is defined as empty array ? Don't we agree to set a default value for it? 

But if your justification is since core-client-rest has already done default logic https://github.com/Azure/azure-sdk-for-js/blob/b39179502e421fef9399be869b9e8c283ada6d57/sdk/core/core-client-rest/src/clientHelpers.ts#L49 and we should do nothing here. I am fine with it. 

Just feel like a little weird to rely on core to set the default value. If customers run into some issues when they forget to pass the scopes and our default value is incorrect, they will have to dig into the core client to find out what's the problem. 

@joheredi how do you think ?

Reply:
IIRC defaulting to "{endpoint}/.default" was a bit optimistic since it is not a standard for Data Plane. I think we can do the following to improve the chances that if no scope was provided by mistake, so that customers can have a better chance of catching a problem:

1. Instead of letting core-client-rest handle the default, do it in the generated code. While the same case is handled in core, having it explicit in the generated code will make it easier for the author and reviewers to catch if `endpoint}/.default` is an invalid scope.
2. Warn during code gen that no scopes were passing and suggest the author to double check the generated default to make sure it is correct.

Reply:
> Instead of letting core-client-rest handle the default, do it in the generated code.

Sure, we could set this default explicitly.

> Warn during code gen that no scopes were passing and suggest the author to double check the generated default to make sure it is correct.

We do have a warning in codegen and you could check the UT here: https://github.com/Azure/autorest.typescript/blob/4697bc247268d88e4ac2b870313adf7280c6e96b/packages/typespec-ts/test/unit/clientFactoryGenerator.spec.ts#L367